Extensions/Leaderboards/leaderboardstools.ts

```diff
@@ -24,12 +25,16 @@ namespace gdjs {
           errorVar.setString('Wait before sending a new score.');
         } else {
           const baseUrl = 'https://api.gdevelop-app.com/play';
+          const game = runtimeScene.getGame();
           fetch(
             `${baseUrl}/game/${gdjs.projectData.properties.projectUuid}/leaderboard/${leaderboardId}/entry`,
             {
               body: JSON.stringify({
                 playerName: playerName,
                 score: score,
+                sessionId: game.getSessionId(),
+                clientPlayerId: game.getPlayerId(),
+                location: (window as any).location.href,
```

Comment:
Mmmh, I just [read on the subject](https://stackoverflow.com/questions/32598971/whats-the-purpose-of-if-typeof-window-undefined) and it could be undefined indeed if the code is not executed in a browser page (maybe if the game is packaged for android as an app ?)
I am not sure what we could send instead and I am not sure if there could be an equivalent in this case

Reply:
That could happen if in a JS environment like Cocos2d-x which embeds a JS engine without a browser.

If window is undefined, you could send an empty string yes!

> maybe if the game is packaged for android as an app ?

Our current solution, Cordova, is embedding a webview as an app, so that won't happen. (but as said, it's right to be robust against this)

Reply:
The goal of having this info is to prevent cheating. Could it be useful to have the platform if the location is undefined?

Reply:
So, I looked at systeminfotools to help me know what could happen if window is not defined.
I found out about Cocos2d (I will call the platform method to get the platform) and about CocoonJS. But I could not find any documentation on the last one. I guess, ultimately, we will send an empty string if there neither window nor Cocos2d

Reply:
@4ian What's your opinion on that?

Reply:
I would let this empty if we don't have the information of the window.location.href (that's sad, but that's the truth, and this is by itself an information) (and we can always add a `platform` or similar field later)